javagen/src/main/java/com/azure/autorest/template/StreamSerializationModelTemplate.java

```diff
@@ -381,7 +381,16 @@ private static void serializeJsonProperty(JavaBlock methodBlock, ClientModelProp
         // This is primitives, boxed primitives, a small set of string based models, and other ClientModels.
         String fieldSerializationMethod = wireType.jsonSerializationMethodCall("jsonWriter", serializedName,
             propertyValueGetter);
-        if (fieldSerializationMethod != null) {
+        if (wireType == ClassType.BINARY_DATA) {
+            // Special handling for BinaryData (instead of using "serializationMethodBase" and "serializationValueGetterModifier")
+            // The reason is that some backend would fail the request on "null" value (e.g. OpenAI)
+            String writeBinaryDataExpr = "jsonWriter.writeUntypedField(\"" + serializedName + "\", " + propertyValueGetter + ".toObject(Object.class));";
+            if (!property.isRequired()) {
+                methodBlock.ifBlock(propertyValueGetter + " != null", ifAction -> ifAction.line(writeBinaryDataExpr));
+            } else {
+                methodBlock.line(writeBinaryDataExpr);
+            }
+        } else if (fieldSerializationMethod != null) {
```

Comment:
Move code branch of `BinaryData` before code branch of `fieldSerializationMethod`, as even when `ClassType.BINARY_DATA` has `fieldSerializationMethod`, it still need special handling here as model property.